File: BubbleSort/bubblesort.lua
function bubble(list)
  local i = 1
  local swap
  
  while (i < #list) do
    if (list[i] > list[i+1]) then
      swap = list[i]
      list[i] = list[i+1]
      list[i+1] = swap
      if i > 1 then i = i - 1 end
    else 
      i = i + 1
    end
  end
  
  return list
end

-- Example Usage:

local sorted = bubble({1, 7, 2, 6, 3, 9, 1, 3, 2, 0})

for i = 1, #sorted do
  print(sorted[i])
end

File: bubble_sort/bubble_sort.c
// C program for implementation of Bubble sort
#include <stdio.h>

void swap(int *xp, int *yp)
{
	int temp = *xp;
	*xp = *yp;
	*yp = temp;
}

// A function to implement bubble sort
void BubbleSort (int a[], int length)
{
	int i, j, temp;
	
    for (i = 0; i < length; i++)
    {
        for (j = 0; j < length - 1; j++)
        {
            if (a[j + 1] < a[j])
            {
                temp = a[j];
                a[j] = a[j + 1];
                a[j + 1] = temp;
            }
        }
    }
}

/* Function to print an array */
void printArray(int arr[], int size)
{
	int i;
	for (i=0; i < size; i++)
		printf("%d ", arr[i]);
	printf("n");
}

// Driver program to test above functions
int main()
{
	int arr[] = {64, 34, 25, 12, 22, 11, 90};
	int n = sizeof(arr)/sizeof(arr[0]);
	bubbleSort(arr, n);
	printf("Sorted array: \n");
	printArray(arr, n);
	return 0;
}

